modules/logger/logger.py

```diff
@@ -143,6 +145,48 @@ def info(self, message: str, log_with_frame_info: bool = True) -> None:
             message = self.message_and_metadata(message, caller_frame)
         self.logger.info(message)
 
+    def save_image(
+        self, image: np.ndarray, filename: str, log_with_frame_info: bool = True
+    ) -> None:
+        """
+        Logs an image.
+
+        Args:
+            image: The image to log.
+            filename: The filename to save the image as.
+            log_with_frame_info: Whether to log the frame info.
+        """
+        img = Image.fromarray(image)
+
+        # Get the file handler to save the image
+        file_handler = next(
+            (
+                handler
+                for handler in self.logger.handlers
+                if isinstance(handler, logging.FileHandler)
+            ),
+            None,
+        )
+        if not file_handler:
+            self.error("No file handler found to save image.")
+            return
+
+        log_dir = os.path.dirname(file_handler.baseFilename)
+
+        img_dir = os.path.join(log_dir, "images")
+        os.makedirs(img_dir, exist_ok=True)
+
+        timestamp = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+        img_path = os.path.join(img_dir, f"{filename}_{timestamp}.png")
```

Comment:
I feel like you don't need to do this, the logger already has a file handler initialized in create. You just need to save an image.